compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/SwingPanel.desktop.kt

```diff
@@ -278,11 +259,54 @@ private class FocusSwitcher<T : Component>(
     }
 }
 
-private class ComponentInfo<T : Component>(
-    container: SwingPanelContainer
+private class SwingInteropComponent<T : Component>(
+    container: SwingPanelContainer,
+    var update: (T) -> Unit
 ): InteropComponent(container) {
-    lateinit var component: T
-    lateinit var updater: Updater<T>
+    private var component: T? = null
+    private var updater: Updater<T>? = null
+
+    fun dispose() {
+        container.remove(component)
+        updater?.dispose()
+        component = null
+        updater = null
+    }
+
+    fun setComponent(component: T) {
```

Comment:
```suggestion
    fun setupComponent(component: T) {
        check(component == null)
```

Otherwise it looks like we can call it 2 times, but the code isn't adapted for that.

Reply:
Renamed to `setupUserComponent`/`cleanUserComponent`. And `SwingInteropComponent.component` to `userComponent` to avoid confusions